spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceProperties.java

```diff
@@ -118,11 +120,17 @@ public void setDynamicDestinations(String[] dynamicDestinations) {
 	public void setApplicationContext(ApplicationContext applicationContext)
 			throws BeansException {
 		this.applicationContext = (ConfigurableApplicationContext) applicationContext;
-		// override the bindings store with the environment-initializing version if in a
-		// Spring context
-		this.bindings = new EnvironmentEntryInitializingTreeMap<>(this.applicationContext,
-				BindingProperties.class, "spring.cloud.stream.default",
-				new TreeMap<String, BindingProperties>(String.CASE_INSENSITIVE_ORDER));
+	}
+
+	@Override
+	public void setEnvironment(Environment environment) {
```

Comment:
This takes into account the possibility that `setApplicationContext()` may be called manually,  (outside a Spring context), e.g. the way it's currently done during test set up. We need the context and the environment for different reasons - if running within a Spring context, they would both be set, otherwise not. This allows not to override the map if there's no actual environment to read defaults from.